src/io/flutter/run/LaunchState.java

```diff
@@ -120,9 +121,21 @@ protected RunContentDescriptor launch(@NotNull ExecutionEnvironment env) throws
     final Project project = getEnvironment().getProject();
     final FlutterDevice device = DeviceService.getInstance(project).getSelectedDevice();
 
+    // If the device is null and the project is not a flutter web project, show a message that a device is required and return null.
```

Comment:
instead of adding a comment, add an isFlutterWeb() method. That way it is self documenting at it will be clear when the hack needs to be removed.